File: Testing/PythonUnitTestingExamples/tests/test_patient_count_summary.py
import numpy as np
import pandas as pd
from ..patient_count_summary import patient_count_summary


def test_patient_count_summary_shape():

    test_df = pd.DataFrame({"PracticeNM":["Practice A", "Practice B", "Practice A", "Practice C"],
                            "MRN":["001", "002", "003", "004"]
    })

    assert patient_count_summary(test_df).shape == (3,2)


def test_patient_count_summary_na_shape():
    test_df = pd.DataFrame({"PracticeNM":["Practice A", "Practice B", "Practice A", np.nan],
                            "MRN":["001", "002", "003", "004"]
    })

    assert patient_count_summary(test_df).shape == (3,2)

def test_patient_count_summary_na_mrn():

    test_df = pd.DataFrame({"PracticeNM":["Practice A", "Practice B", "Practice A", "Practice C"],
                            "MRN":["001", "002", "003", np.nan]
    })

    expected_output_df = pd.DataFrame({"PracticeNM":["Practice A", "Practice B", "Practice C"],
                            "PatientCNT":[2, 1, 0]
    })

    df = patient_count_summary(test_df)

    assert pd.testing.assert_frame_equal(df, expected_output_df) is None

File: Testing/RUnitTestingExamples/tests/test_patient_count_summary.R
source("../patient_count_summary.R", chdir = TRUE)
library(testthat)


test_df <- data.frame(
  PracticeNM = c("Practice A", "Practice B", "Practice A", "Practice C"),
  MRN = c("001", "002", "003", "004")
)

test_that("patient_count_summary dim", {
  expect_equal(dim(patient_count_summary(test_df)), c(3, 2))
})


test_practice_na_df <- data.frame(
  PracticeNM = c("Practice A", "Practice B", "Practice A", NA),
  MRN = c("001", "002", "003", "004")
)

test_that("patient_count_summary NA dim", {
  expect_equal(dim(patient_count_summary(test_practice_na_df)), c(3, 2))
})


test_mrn_na_df <- data.frame(
  PracticeNM = c("Practice A", "Practice B", "Practice A", "Practice C"),
  MRN = c("001", "002", "003", NA)
)

expected_output_df <- data.frame(
  PracticeNM = c("Practice A", "Practice B", "Practice C"),
  PatientCNT = c(2, 1, 0)
)

test_that("patient_count_summary NA MRN", {
  expect_equal(patient_count_summary(test_mrn_na_df), expected_output_df)
})
